tsdb/wlog/watcher.go

```diff
@@ -290,16 +295,9 @@ func (w *Watcher) Run() error {
 	for !isClosed(w.quit) {
 		w.currentSegmentMetric.Set(float64(currentSegment))
 
-		// Re-check on each iteration in case a new segment was added,
-		// because watch() will wait for notifications on the last segment.
-		_, lastSegment, err := w.firstAndLast()
-		if err != nil {
-			return fmt.Errorf("wal.Segments: %w", err)
-		}
-		tail := currentSegment >= lastSegment
-
-		level.Debug(w.logger).Log("msg", "Processing segment", "currentSegment", currentSegment, "lastSegment", lastSegment)
```

Comment:
nit: it may be helpful to keep the debug log `level.Debug(w.logger).Log("msg", "Processing segment", "currentSegment", currentSegment)`

Reply:
I agree the line can be useful; `lastSegment` will be frozen at its first value which is potentially confusing.